src/mappings/StakingRewards/shared.ts

```diff
@@ -0,0 +1,234 @@
+  stakingRewards.periodFinish = contract.periodFinish().toI32()
+  stakingRewards.rewardRate = contract.rewardRate()
+  stakingRewards.totalStakingRewards = getTotalStakingRewards(stakingRewards)
+
+  if (isWithPlatformToken(type)) {
+    stakingRewards.platformRewardRate = (contract as StakingRewardsWithPlatformToken).platformRewardRate()
+    stakingRewards.totalPlatformRewards = getTotalPlatformRewards(
+      stakingRewards,
+    )
+  }
+
+  stakingRewards.save()
+}
+
+export function handleStakedForType<C>(
+  contractAddress: Address,
+  type: StakingRewardsContractType,
+  user: Address,
+): void {
+  let contract = getContractForType<C>(contractAddress, type)
+
+  {
+    let stakingRewards = getOrCreateStakingRewardsContract(
+      contractAddress,
+      type,
+    )
+
+    stakingRewards.lastUpdateTime = contract.lastUpdateTime().toI32()
+    stakingRewards.rewardPerTokenStored = contract.rewardPerTokenStored()
+    stakingRewards.periodFinish = contract.periodFinish().toI32()
+    stakingRewards.rewardRate = contract.rewardRate()
+    stakingRewards.totalStakingRewards = getTotalStakingRewards(stakingRewards)
+
+    if (isWithPlatformToken(type)) {
+      stakingRewards.platformRewardPerTokenStored = (contract as StakingRewardsWithPlatformToken).platformRewardPerTokenStored()
+      stakingRewards.totalPlatformRewards = getTotalPlatformRewards(
+        stakingRewards,
+      )
+    }
+
+    stakingRewards.save()
+  }
+
+  {
+    let reward = getOrCreateStakingReward(
+      contractAddress,
+      user,
+      StakingRewardType.REWARD_PER_TOKEN_PAID,
+    )
+    reward.amount = contract.userRewardPerTokenPaid(user)
+    reward.save()
+  }
+
+  if (isWithPlatformToken(type)) {
+    let reward = getOrCreateStakingReward(
+      contractAddress,
+      user,
+      StakingRewardType.PLATFORM_REWARD_PER_TOKEN_PAID,
+    )
+    reward.amount = (contract as StakingRewardsWithPlatformToken).userPlatformRewardPerTokenPaid(
+      user,
+    )
+    reward.save()
+  }
+}
+
+export function handleWithdrawnForType<C>(
+  contractAddress: Address,
+  type: StakingRewardsContractType,
+  user: Address,
+): void {
+  let contract = getContractForType<C>(contractAddress, type)
+
+  {
+    let stakingRewards = getOrCreateStakingRewardsContract(
+      contractAddress,
+      StakingRewardsContractType.STAKING_REWARDS,
+    )
+
+    stakingRewards.lastUpdateTime = contract.lastUpdateTime().toI32()
+    stakingRewards.rewardPerTokenStored = contract.rewardPerTokenStored()
+    stakingRewards.totalStakingRewards = getTotalStakingRewards(stakingRewards)
+
+    if (isWithPlatformToken(type)) {
+      stakingRewards.platformRewardPerTokenStored = (contract as StakingRewardsWithPlatformToken).platformRewardPerTokenStored()
+      stakingRewards.totalPlatformRewards = getTotalPlatformRewards(
+        stakingRewards,
+      )
+    }
+
+    stakingRewards.save()
+  }
+
+  {
+    let reward = getOrCreateStakingReward(
```

Comment:
This 'rewardperTokenPaid' should be paired with the 'rewards' prop, instead of being seperate. They are used in combination with the rest of the data to calculate the current pro rata reward